mcuboot: specifying -DCONF_FILE results in failure

**Describe the bug**
Specifying explicitly mcuboot configuration with -DCONF_FILE=/path/to/some_config.conf results in:
```
-- Found assembler: /opt/zephyr-sdk/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMake Error at CMakeLists.txt:235 (get_filename_component):
  get_filename_component called with incorrect number of arguments


-- Configuring incomplete, errors occurred!
```
It turns out that CONF_FILE in mcuboot's CMakeLists.txt:235 **is** set normally when -DCONF_FILE **is not** used, and **is not** set when -DCONF_FILE **is** passed to cmake in configuration phase. So just the opposite what one could initially expect. I think this has to do with recent added line in Zephyr `unset(CONF_FILE CACHE)`, but didn't check that further.

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -DCONF_FILE=prj.conf ../bootloader/mcuboot/boot/zephyr/
3. See error

**Expected behavior**
Should work same as using prj.conf implicitly with `cmake -GNinja -DBOARD=nrf52840dk_nrf52840 ../bootloader/mcuboot/boot/zephyr/`.

**Impact**
Prevents using custom mcuboot configuration (e.g. differente key type, smaller config, etc.), which is a big problem.

**Logs and console output**
```
[build@zm build]$ cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -DCONF_FILE=prj.conf ../bootloader/mcuboot/boot/zephyr/Including boilerplate (Zephyr base): /build/work/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /build/work/zephyrproject/bootloader/mcuboot/boot/zephyr
-- Zephyr version: 2.4.0-rc1 (/build/work/zephyrproject/zephyr)
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.5") found components: Interpreter 
-- Found west (found suitable version "0.7.2", minimum required is "0.7.1")
-- Board: nrf52840dk_nrf52840
-- Cache files will be written to: /build/.cache/zephyr
-- Found toolchain: zephyr (/opt/zephyr-sdk)
-- Found dtc: /opt/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /build/work/zephyrproject/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts
-- Found devicetree overlay: /build/work/zephyrproject/bootloader/mcuboot/boot/zephyr/dts.overlay
-- Generated zephyr.dts: /build/work/zephyrproject/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /build/work/zephyrproject/build/zephyr/include/generated/devicetree_unfixed.h
Parsing /build/work/zephyrproject/bootloader/mcuboot/boot/zephyr/Kconfig
Loaded configuration '/build/work/zephyrproject/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig'
Merged configuration 'prj.conf'
Configuration saved to '/build/work/zephyrproject/build/zephyr/.config'
Kconfig header saved to '/build/work/zephyrproject/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/zephyr-sdk/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMake Error at CMakeLists.txt:235 (get_filename_component):
  get_filename_component called with incorrect number of arguments


-- Configuring incomplete, errors occurred!
See also "/build/work/zephyrproject/build/CMakeFiles/CMakeOutput.log".
See also "/build/work/zephyrproject/build/CMakeFiles/CMakeError.log".
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.11.2
 - 2.4.0-rc1

--- a/cmake/app/boilerplate.cmake
+++ b/cmake/app/boilerplate.cmake
@@ -469,7 +469,11 @@ please check your installation. ARCH roots searched: \n\
 ${ARCH_ROOT}")
 endif()
 
-if(CONF_FILE)
+if(DEFINED CONF_FILE)
+  # This ensures that CACHE{CONF_FILE} will be set correctly to current scope
+  # variable CONF_FILE. An already current scope variable will stay the same.
+  set(CONF_FILE ${CONF_FILE})
+
   # CONF_FILE has either been specified on the cmake CLI or is already
   # in the CMakeCache.txt. This has precedence over the environment
   # variable CONF_FILE and the default prj.conf

